pkg/devicerepository/config.go

```diff
@@ -0,0 +1,48 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package devicerepository
+
+import (
+	"context"
+
+	"go.thethings.network/lorawan-stack/v3/pkg/config"
+	"go.thethings.network/lorawan-stack/v3/pkg/devicerepository/store"
+)
+
+// Config represents the DeviceRepository configuration.
+type Config struct {
+	Store StoreConfig `name:"store"`
+
+	ConfigSource string                `name:"config-source" description:"Source of the Device Repository (static, directory, url, blob)"`
```

Comment:
I thought I already commented on this (maybe it's in one of the other pull requests), but since we're not violating compatibility commitment by changing unused and undocumented config, I think we can also write `Source` instead of `ConfigSource`, since this is about the source of the device repository, not the source of the config of the device repository.

Reply:
Yes, this was first suggested in #3543 and fixed there.